examples/02_decoding/plot_haxby_anova_svm.py

```diff
@@ -14,120 +14,91 @@
-# Prepare the fMRI data: smooth and apply the mask
-# -------------------------------------------------
-from nilearn.input_data import NiftiMasker
-
+# ANOVA pipeline with :class:`nilearn.decoding.Decoder` object
+# ------------------------------------------------------------
+#
+# Nilearn Decoder object aims to provide smooth user experience by acting
+# as a pipeline of several tasks: preprocessing with NiftiMasker, decoding
+# with different types of estimators (in this example is Support Vector
+# Machine with a linear kernel) on nested cross-validation, selecting
+# relevant features with ANOVA -- a classical univariate feature selection
+# based on F-test.
+from nilearn.decoding import Decoder
+# Here screening_percentile is set to 5 percent
 mask_filename = haxby_dataset.mask
-
-# For decoding, standardizing is often very important
-# note that we are also smoothing the data
-masker = NiftiMasker(mask_img=mask_filename, smoothing_fwhm=4,
-                     standardize=True, memory="nilearn_cache", memory_level=1)
-func_filename = haxby_dataset.func[0]
-X = masker.fit_transform(func_filename)
-# Apply our condition_mask
-X = X[condition_mask]
-
-#############################################################################
-# Build the decoder
-# ------------------
-# Define the prediction function to be used.
-# Here we use a Support Vector Classification, with a linear kernel
-from sklearn.svm import SVC
-svc = SVC(kernel='linear')
-
-# Define the dimension reduction to be used.
-# Here we use a classical univariate feature selection based on F-test,
-# namely Anova. When doing full-brain analysis, it is better to use
-# SelectPercentile, keeping 5% of voxels
-# (because it is independent of the resolution of the data).
-from sklearn.feature_selection import SelectPercentile, f_classif
-feature_selection = SelectPercentile(f_classif, percentile=5)
-
-# We have our classifier (SVC), our feature selection (SelectPercentile),and now,
-# we can plug them together in a *pipeline* that performs the two operations
-# successively:
-from sklearn.pipeline import Pipeline
-anova_svc = Pipeline([('anova', feature_selection), ('svc', svc)])
+decoder = Decoder(estimator='svc', mask=mask_filename, smoothing_fwhm=4,
+                  standardize=True, screening_percentile=5, scoring='accuracy')
 
 #############################################################################
 # Fit the decoder and predict
 # ----------------------------
-anova_svc.fit(X, conditions)
-y_pred = anova_svc.predict(X)
+decoder.fit(func_filename, conditions)
+y_pred = decoder.predict(func_filename)
 
 #############################################################################
 # Obtain prediction scores via cross validation
 # -----------------------------------------------
-from sklearn.model_selection import LeaveOneGroupOut, cross_val_score
-
-# Define the cross-validation scheme used for validation.
```

Comment:
Could you point to the decoding_intro docpage in a topic that explains the cross-validation process ?